pkg/cedar/core/client.go

```diff
@@ -154,3 +158,123 @@ func (c *Client) GetSystem(ctx context.Context, id string) (*models.CedarSystem,
 		SystemMaintainerOrgComp: responseArray[0].SystemMaintainerOrgComp,
 	}, nil
 }
+
+// GetDeploymentsOptionalParams represents the optional parameters that can be used to filter deployments
+// when searching through the CEDAR API
+// TODO - is this how we want to handle multiple optional parameters?
+type GetDeploymentsOptionalParams struct {
+	DeploymentType null.String
+	State          null.String
+	Status         null.String
+}
+
+// GetDeployments makes a GET call to the /deployment endpoint
+func (c *Client) GetDeployments(ctx context.Context, systemID string, optionalParams *GetDeploymentsOptionalParams) ([]*models.CedarDeployment, error) {
+	if !c.cedarCoreEnabled(ctx) {
+		appcontext.ZLogger(ctx).Info("CEDAR Core is disabled")
+		return []*models.CedarDeployment{}, nil
+	}
+
+	// Construct the parameters
+	params := apideployments.NewDeploymentFindListParams()
+	params.SetSystemID(systemID)
+	params.HTTPClient = c.hc
+
+	if optionalParams != nil {
+		if optionalParams.DeploymentType.Ptr() != nil {
+			params.SetDeploymentType(optionalParams.DeploymentType.Ptr())
+		}
+
+		if optionalParams.State.Ptr() != nil {
+			params.SetState(optionalParams.State.Ptr())
+		}
+
+		if optionalParams.Status.Ptr() != nil {
+			params.SetStatus(optionalParams.Status.Ptr())
+		}
+	}
+
+	// Make the API call
+	resp, err := c.sdk.Deployment.DeploymentFindList(params, c.auth)
+	if err != nil {
+		return []*models.CedarDeployment{}, err
+	}
+
+	if resp.Payload == nil {
+		return []*models.CedarDeployment{}, fmt.Errorf("no body received")
+	}
+
+	responseArray := resp.Payload.Deployments
+
+	if len(responseArray) == 0 {
+		return nil, &apperrors.ResourceNotFoundError{Err: fmt.Errorf("no deployments found"), Resource: []*models.CedarDeployment{}}
+	}
+
+	// Convert the auto-generated struct to our own pkg/models struct
+	retVal := []*models.CedarDeployment{}
+
+	// Populate the Deployment field by converting each item in resp.Payload.Deployments
+	// generated swagger client turns JSON nulls into Go zero values, so use null/zero package to convert them back to nullable values
+	for _, deployment := range resp.Payload.Deployments {
+		if deployment.ID == nil {
```

Comment:
Not sure about these checks; the CEDAR docs indicate that a deployment's ID/Name/SystemID should never be null. Doing these checks and failing to decode the deployment if any of these fields aren't present means that we guarantee their presence in our code that uses the CEDAR client, but it does mean that we're not really resilient in the presence of CEDAR errors. Would appreciate thoughts.